fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -2376,6 +2411,11 @@ ACTOR Future<Void> reply(CommitBatchContext* self) {
 
 	state const Optional<UID>& debugID = self->debugID;
 
+	if (SERVER_KNOBS->SHARD_THROTTLING_ENABLED && self->throttledShard && deterministicRandom()->coinflip()) {
```

Comment:
This doesn't seem right. The idea should be introducing a new retryable error and send that error back to the client, like `tr.reply.sendError(transaction_too_old());` later in this function.


The client, at NativeAPI.actor.cpp, should perform exponential backoff.

Reply:
:+1: